test/powershell/engine/ExperimentalFeature/ExperimentalFeature.Basic.Tests.ps1

```diff
@@ -356,17 +361,69 @@ Describe "Experimental Feature Basic Tests - Feature-Enabled" -Tag "CI" {
 
         $command = Get-Command $Name
         $command.CommandType | Should -Be $CommandType
-        ## 11 common parameters + '-Name' (dynamic parameters are not triggered)
-        $command.Parameters.Count | Should -Be 12
+        ## Common parameters + '-Name' (dynamic parameters are not triggered)
+        $command.Parameters.Count | Should -Be ($CommonParameterCount + 1)
         $command.Parameters["Name"] | Should -Not -BeNullOrEmpty
 
         $command = Get-Command $Name -ArgumentList "Joe"
-        ## 11 common parameters + '-Name' and '-ConfigFile' (dynamic parameters are triggered)
-        $command.Parameters.Count | Should -Be 13
+        ## Common parameters + '-Name' and '-ConfigFile' (dynamic parameters are triggered)
+        $command.Parameters.Count | Should -Be ($CommonParameterCount + 2)
         $command.Parameters["ConfigFile"].Attributes.Count | Should -Be 2
         $command.Parameters["ConfigFile"].Attributes[0] | Should -BeOfType [parameter]
         $command.Parameters["ConfigFile"].Attributes[1] | Should -BeOfType [ValidateNotNullOrEmpty]
 
         $command.Parameters.ContainsKey("ConfigName") | Should -Be $false
     }
 }
+
+Describe "Expected errors" -Tag "CI" {
+    It "'[Experimental()]' should fail to construct the attribute" {
+        { [Experimental()]param() } | Should -Throw -ErrorId "MethodCountCouldNotFindBest"
+    }
+
+    It "Argument validation for constructors of 'ExperimentalAttribute' and 'ParameterAttribute'" {
+        { [Experimental]::new("", "None") } | Should -Throw -ErrorId "PSArgumentNullException"
```

Comment:
Would be better to have these as TestCases so that the first failure doesn't prevent the rest from running

Reply:
Will update the test.